File: bitcoin-rpc-client-java/src/main/java/io/github/anttikaikkonen/bitcoinrpcclientjava/models/ScriptPubKey.java
package io.github.anttikaikkonen.bitcoinrpcclientjava.models;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class ScriptPubKey {
    
    private String asm;
    private String hex;
    private int reqSigs;
    private String type;
    private String[] addresses;
    
}

File: blockchain-graphql-server/src/models/scriptpubkey.ts
import { ObjectType, Field, Int } from "type-graphql";
import { Address } from "./address";

@ObjectType()
export class ScriptPubKey {
    @Field({nullable: false, complexity: 1})
    asm: string;
    @Field({nullable: false, complexity: 1})
    hex: string;
    @Field(type => Int, {nullable: false, complexity: 1})
    reqSigs: number;
    @Field({nullable: false, complexity: 1})
    type: string;
    @Field(type => [Address], {nullable: true, complexity: 1})
    addresses: Address[];

}